stackrox/stackrox
# ROX-9527: Create actual ECR autogenerated integrations

## Description

Changes the autogenerated image integration for ECR from regular Docker integration to an actual ECR integration, effectively moving it to the ECR Integrations tab.

This allows better visibility for customers on how ECR-based images are being accessed and provide the opportunity for customers to modify these integrations, specially if the instance role used in their nodes does not have permissions to use the registry.

The change touches both Sensor and Central:

- In Sensor, it changes the current AWS credentials manager to send and ECR integration event, rather than a Docker integration event. New fields were required to the event to allow the Authentication Data to be passed along in the message.
- In Central, it changes the image integration pipeline to process ECR integration events, as well as it changes how the ECR integration retrieves its Docker Credentials, never refreshing them if the integration was autogenerated and contains static authentication data.

## Checklist

Documentation update PR: https://github.com/openshift/openshift-docs/pull/44071

## Testing Performed

1. Deployed ACS to a EKS cluster at account `044192981154` IAM user `jvdm`.
2. Go to "Vulnerability Management/Dashboard/Images" and verify some images deployed by EKS from ECR don't have vulnerability data associated with them.
3. Setup `kubectl`.
4. `kubectl -n stackrox set env deploy/sensor ROX_ECR_AUTO_INTEGRATION=true`, to enable the feature this PR is supporting.
5. Go to "Integrations/ECR" and check integrations were created for the images deployed by EKS. The name should be "Autogenerated AWS account 602401143452 and region us-west-2 for cluster remote".
6. Deploy a pod from ECR. You can use `kubectl run --rm -i --tty cporter --image 044192981154.dkr.ecr.us-west-2.amazonaws.com/cporter-ecr:1.6 --restart=Never --image-pull-policy="Always" -- /bin/bash`.
7. While the pod is running, go to "Integrations/ECR" and check another integration was created for 044192981154.
8. Go to "Vulnerability Management/Dashboard/Images" and verify the ECR-based images have vulnerability, except an internal one which the IAM instance role doesn't have access to, called `602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/coredns:v1.8.4-eksbuild.1`.
9. Go to "Integrations/ECR", and modify one of the integrations, test, and save. Re-do the same steps using AWS creds for `jvdm` or any other IAM user of your choice tha thas access to the registry, or using the "Use Instance Role" flag.
